napari/layers/_multiscale_data.py

```diff
@@ -0,0 +1,121 @@
+from __future__ import annotations
+
+from typing import List, Optional, Sequence, Tuple, Union
+
+import numpy as np
+
+from ._data_protocols import LayerDataProtocol, assert_protocol
+from .utils.layer_utils import compute_multiscale_level_and_corners
+
+
+class MultiScaleData(Sequence[LayerDataProtocol], LayerDataProtocol):
```

Comment:
Should this really explicitly extend `LayerDataProtocol`? In the spirit of structural sub-typing, isn't it enough to define the methods in the `LayerDataProtocol`?

Whereas `Sequence` is an `ABC` right? Or is it in `collections.abc` just to confuse me?

Reply:
> isn't it enough to define the methods in the LayerDataProtocol?

yep, I can remove it.  In this case it would have the same exact result.  It's not "gaining" anything from inheriting from `LayerDataProtocol` (other than being a bit self-documenting), but it's also not turning this into a `typing.Protocol` (for that I would need to explicitly inherit from protocol here too).

Reply:
My only concern is that I think `isinstance` will pass even if `MultiScaleData` doesn't actually implement `LayerDataProtocol`, which nullifies some of the benefits of this PR. To take a trivial example:

```python
In [27]: @typing.runtime_checkable
    ...: class MyProtocol(typing.Protocol):
    ...:     def foo(self, x: int) -> int:
    ...:         pass
    ...: 

In [28]: class MyClass(MyProtocol):
    ...:     pass
    ...: 

In [29]: isinstance(MyClass(), MyProtocol)
Out[29]: True
```

Reply:
while that example is true, I think it's a little contrived, because in your `foo` is a no-op and `MyClass` doesn't re-implement it, whereas `MultiScaleData` _does_ re-implement all of those methods.

If one was `really` making a Protocol that was intended to be subclassed with safety, they should use `@abstractclass` (`protocol.Protocol` is actually a [`abc.ABCMeta`](https://docs.python.org/3/library/abc.html#abc.ABCMeta) and so can use the `@abstractmethod` decorator).

but the point is taken: the current code makes it look like something is potentially gained from subclassing, and it isn't (since all of the structural typing classes are indeed already overwritten there).  will remove